lib/engines/content_block_manager/lib/tasks/delete_content_block.rake

```diff
@@ -0,0 +1,24 @@
+namespace :content_block_manager do
+  desc "Delete content block"
+  task :delete_content_block, [:content_id] => :environment do |_t, args|
+    content_id = args[:content_id]
+    content_block_document = ContentBlockManager::ContentBlock::Document.find_by(content_id:)
+
+    abort("A content block with the content ID `#{content_id}` cannot be found") unless content_block_document
+
+    @host_content_items = ContentBlockManager::HostContentItem.for_document(content_block_document)
+
+    abort("Content block `#{content_id}` cannot be deleted because it has host content. Try removing the dependencies and trying again") unless @host_content_items.items.count.zero?
+
+    Services.publishing_api.unpublish(
```

Comment:
can't quite figure out what it does on Publishing API at first glance - wondering if it will affect this query, whether we need to add some kind of check that it hasn't been unpublished? https://github.com/alphagov/publishing-api/pull/3223/files#diff-f75e6df9ef6e0a772a32e0f688a7796a31b1a892547b49210304a1cd8a5545ed

Reply:
It sets all editions to `unpublished`, so this part of the query should cover it https://github.com/alphagov/publishing-api/blob/6254843f121efb9e17e559fe58c09daf963356e9/app/queries/get_embedded_editions_from_host_edition.rb#L10